marklogic-client-api/src/test/java/com/marklogic/client/test/LineSplitterTest.java

```diff
@@ -0,0 +1,100 @@
+package com.marklogic.client.test;
+
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.DatabaseClientFactory;
+import com.marklogic.client.datamovement.LineSplitter;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.StringHandle;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+import java.util.zip.GZIPInputStream;
+
+import static org.junit.Assert.*;
+
+public class LineSplitterTest {
+    static final private String jsonlFile = "src/test/resources/data" + File.separator + "line-delimited.jsonl";
+    static final private String jsonlGzipFile = "src/test/resources/data" + File.separator + "line-delimited.jsonl.gz";
+    static final private String xmlFile = "src/test/resources/data" + File.separator + "line-delimited.txt";
+    private DatabaseClient client;
+
+    @Before
+    public void setUp() {
+        client = DatabaseClientFactory.newClient("localhost", 8012,
+                new DatabaseClientFactory.DigestAuthContext("rest-admin", "x"));
+    }
+
+    @Test
+    public void testSplitter() throws Exception {
+        LineSplitter splitter = new LineSplitter();
+        Stream<StringHandle> contentStream = splitter.split(new FileInputStream(jsonlFile));
+        assertNotNull(contentStream);
+
+        StringHandle[] result = contentStream.toArray(size -> new StringHandle[size]);
+        assertNotNull(result);
+        assertEquals(result.length, Files.lines(Paths.get(jsonlFile)).count());
```

Comment:
The test might run slightly faster if, instead of reading the file twice, the test moved the originalResult up to line 37 as an initialization and used the originalResult.length for this assertion.  Just a thought.

Also, if we have asserted result.length is equal to originalResult.length, we don't need to test both in the loop below.  Just a thought.

Reply:
That's true! Thanks!